core/src/text/textBuffer.cpp

```diff
@@ -15,6 +15,22 @@ TextBuffer::TextBuffer(std::shared_ptr<VertexLayout> _vertexLayout)
 TextBuffer::~TextBuffer() {
 }
 
+std::vector<TextBuffer::WordBreak> TextBuffer::findWordBreaks(const std::string& _text) {
```

Comment:
Yep, either if it was in the source data, or added in the stylesheet by a JS function, e.g. `text_source: function(){ return feature.name.replace(' ', '\n') }`. When there are explicit line breaks, we assume that they were desired, so we always break on them.